File: WebApp/src/app/shared/models/catalogue/catUnit.model.ts
export class Unit {
     id: number = 0;
     code: string = '';
     unitNameVn: string = '';
     unitNameEn: string = '';
     unitType: string = '';
     unitTypeName: string = '';
     userCreated: string = '';
     datetimeCreated: any = '';
     userModified: string = '';
     datetimeModified: any = '';
     active: boolean = true;
     inactiveOn: any = '';
     descriptionEn: string = '';
     descriptionVn: string = '';

     constructor(data?: any) {
          const self = this;
          for (const key in data) {
               if (self.hasOwnProperty(key)) {
                    self[key] = data[key];
               }
          }
     }
}

File: WebAPI/eFMS.API.SystemWeb/eFMS.API.Catalogue/eFMS.API.Catalogue.DL/Models/CatUnitModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace eFMS.API.Catalogue.DL.Models
{
    public class CatUnitModel
    {
        public short Id { get; set; }
        [Required]
        public string Code { get; set; }
        [Required]
        public string UnitNameVn { get; set; }
        [Required]
        public string UnitNameEn { get; set; }
        [Required]
        public string UnitType { get; set; }
        public string DescriptionEn { get; set; }
        public string DescriptionVn { get; set; }
        public string UserCreated { get; set; }
        public DateTime? DatetimeCreated { get; set; }
        public string UserModified { get; set; }
        public DateTime? DatetimeModified { get; set; }
        public bool? Active { get; set; }
        public DateTime? InActiveOn { get; set; }
    }
}
